### PR TITLE
Add placeholder for Document Collections

### PR DESCRIPTION
**Pull request by boffbowsh**
Add a specific placeholder for Document Collections that pushes the
content ids of its linked documents to the publishing api. This will be
used to determine which collections an edition is part of for migrated
formats.

Trello: https://trello.com/c/UcqLRRSu/387-add-document-collection-documents-to-placeholder

### CODE REVIEW

**Comment by fofr** [comment]
Should this also include a data migration to publish those placeholders?

(I believe the theory on recent detailed guides missing content items is that a placeholder was added but we never ran a data migration job to publish them all)

**Comment by boffbowsh** [comment]
Yes. Added.

**Comment by fofr** [comment]
Do collections not already have placeholders? Is that from collections-publisher?
https://www.gov.uk/api/content/government/collections/overseas-living-in-guides

**Comment by boffbowsh** [comment]
Whitehall Document Collections (`/government/collections`) already had placeholders, but they didn't include the documents in the placeholders. These are different from the content published by collections-publisher.

File: test/unit/presenters/publishing_api_presenters/document_collection_placeholder_test.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+class PublishingApiPresenters::DocumentCollectionPlaceholderTest < ActiveSupport::TestCase
+  def present(model_instance, options = {})
+    PublishingApiPresenters::DocumentCollectionPlaceholder.new(model_instance, options)
+  end
+
+  test 'presents a placeholder_document_collection ready for adding to the publishing API' do
+    publications_in_collection = [
+      create(:published_policy_paper, title: 'Free bunnies for all citizens'),
+      create(:published_policy_paper, title: 'Photos of cats to be included in all HMRC letters')
```
Comment (fofr) [inline test/unit/presenters/publishing_api_presenters/document_collection_placeholder_test.rb]: 😹

**Comment by fofr** [comment]
:+1:

**Pull request merged**